templates/puppet.spec

```diff
@@ -0,0 +1,52 @@
+%{!?upstream_version: %global upstream_version %{version}%{?milestone}}
+{% if metadata.from_puppetlabs -%}
+%%define upstream_name {{ metadata.puppetlabs_name }}
+{% endif %}
+Name:                   {{ metadata.name }}
+Version:                XXX
+Release:                XXX
+Summary:                {{ metadata.summary }}
+License:                {{ metadata.license }}
+
+URL:                    {{ metadata.project_page }}
+
+Source0:                {{ metadata.source0 }}
+
+BuildArch:              noarch
+
+{% for dep in metadata.dependencies -%}
+Requires:               {{ dep.name }}
+{% endfor %}
+Requires:               puppet >= 2.7.0
+
+%description
+{{ metadata.description }}
+
+%prep
+%setup -q -n {{ '%{'+metadata.upstream_name+'}-%{upstream_version}' }}
+
+find . -type f -name ".*" -exec rm {} +
+find . -size 0 -exec rm {} +
+find . \( -name "*.pl" -o -name "*.sh"  \) -exec chmod +x {} +
+find . \( -name "*.pp" -o -name "*.py"  \) -exec chmod -x {} +
+find . \( -name "*.rb" -o -name "*.erb" \) -exec chmod -x {} +
+find . \( -name spec -o -name ext \) | xargs rm -rf
+
+%build
+
+
+%install
+rm -rf %{buildroot}
+install -d -m 0755 %{buildroot}/%{_datadir}/openstack-puppet/modules/{{ metadata.project }}/
+cp -rp * %{buildroot}/%{_datadir}/openstack-puppet/modules/{{ metadata.project }}/
+{% if "nova" == metadata.project -%}
```

Comment:
I'm not sure about this, it may get crowded over time as some modules have their own particularities. I don't have a great idea for that either.